From: Adam Jackson <ajax@redhat.com>
Date: Sun, 28 Oct 2007 09:37:52 +0100
Subject: [PATCH] Add selinux awareness.

Mesa contains some codegen paths that check for selinux, so we need to
link against libselinux to make that work.

---
--- xserver-20070223/hw/xfree86/dixmods/Makefile.am.selinux-awareness	2007-02-23 11:20:30.000000000 -0500
+++ xserver-20070223/hw/xfree86/dixmods/Makefile.am	2007-02-23 11:24:32.000000000 -0500
@@ -42,7 +42,7 @@
            -I$(top_srcdir)/GL/glx
 
 libGLcore_la_LDFLAGS = -avoid-version
-libGLcore_la_LIBADD = $(top_builddir)/GL/mesa/libGLcore.la
+libGLcore_la_LIBADD = $(top_builddir)/GL/mesa/libGLcore.la -lselinux
 libGLcore_la_SOURCES = GLcoremodule.c
 
 libafb_la_LDFLAGS = -avoid-version
--- xserver-20070223/configure.ac.selinux-awareness	2007-02-23 11:24:31.000000000 -0500
+++ xserver-20070223/configure.ac	2007-02-23 11:25:38.000000000 -0500
@@ -237,6 +237,8 @@
 	;;
 esac
 
+SYS_LIBS="${SYS_LIBS} -lselinux"
+
 dnl BSD *_video.c selection
 AM_CONDITIONAL(ALPHA_VIDEO, [test "x$ALPHA_VIDEO" = xyes])
 AM_CONDITIONAL(ARM_VIDEO, [test "x$ARM_VIDEO" = xyes])
@@ -1010,7 +1012,7 @@
 AM_CONDITIONAL(XNEST, [test "x$XNEST" = xyes])
 
 if test "x$XNEST" = xyes; then
-	XNEST_LIBS="$FB_LIB $MI_LIB $FIXES_LIB $XEXT_LIB $DBE_LIB $XTRAP_LIB $RECORD_LIB $GLX_LIBS $RENDER_LIB $RANDR_LIB $DAMAGE_LIB $MIEXT_DAMAGE_LIB $MIEXT_SHADOW_LIB $XI_LIB $XKB_LIB $XKB_STUB_LIB $COMPOSITE_LIB $XPSTUBS_LIB $CWRAP_LIB $OS_LIB $LIBS"
+	XNEST_LIBS="$FB_LIB $MI_LIB $FIXES_LIB $XEXT_LIB $DBE_LIB $XTRAP_LIB $RECORD_LIB $GLX_LIBS $RENDER_LIB $RANDR_LIB $DAMAGE_LIB $MIEXT_DAMAGE_LIB $MIEXT_SHADOW_LIB $XI_LIB $XKB_LIB $XKB_STUB_LIB $COMPOSITE_LIB $XPSTUBS_LIB $CWRAP_LIB $OS_LIB $LIBS $XSERVER_LIBS"
 	AC_SUBST([XNEST_LIBS])
 fi
 
